src/bin/add/main.rs

```diff
@@ -143,6 +147,34 @@ fn handle_add(args: &Args) -> Result<()> {
     }
     let deps = &args.parse_dependencies()?;
 
+    if let Some(ref requested_features) = args.features {
+        let available_features = deps
+            .iter()
+            .map(|d| d.available_features.clone())
+            .flatten()
+            .collect::<Vec<String>>();
```

Comment:
Won't this mean that if you add multiple deps, if the depenmdency is only valid for one, we'll go along rather than error?

Seems like we need to do this check per-dependency.

Reply:
it would but adding multiple crates with features is disalowed by the CLI so once we have features we know there is only one crate.

Reply:
Looks like we error out if someone sets features with multiple dependencies

We can either
- Remove that check, relying on checking if all of them have the dependency as I mentioned above
- Have an `assert!(deps.len <= 1)` here (assuming this comes after that error check) and then just `deps[8]` rather than looping over each dependency

Reply:
Went for the second approach as I don't know why that check/requirement was added in the first place so I'd rather just leave it until we have a good reason